state/firewallrules.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/juju/errors"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+	"gopkg.in/mgo.v2/txn"
+)
+
+// FirewallRule instances describe the ingress networks
+// whitelist/blacklist for a given well known service.
+// Primarily, whitelisting allowed ingress subnets is the
+// main use case. However, blacklisting subnets is useful
+// to allow restrictions placed on incoming traffic for
+// cross model relations, where the source of traffic is
+// requested from the consuming side.
+// WellKnownService is either a well known internet service
+// (currently just SSH) or a Juju defined value.
+// Supported values are:
+// - ssh
+// - juju-controller
+// - juju-application-offer
+type FirewallRule interface {
```

Comment:
Why not just pass around struct values? I don't see any benefit to making this an interface.

Reply:
Ok, changed to structs.